tests/Feature/FetchProjectStatsTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Maintainer;
+use App\Models\Project;
+use Github\Api\Issue;
+use Github\Api\PullRequest;
+use Github\Api\Repo;
+use GrahamCampbell\GitHub\Facades\GitHub as GitHubClient;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Http;
+use Mockery;
+use Tests\TestCase;
+
+class FetchProjectStatsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function fetch_stats_and_persist_successfully(): void
+    {
+        Http::preventStrayRequests();
+
+        Http::fake([
+            'packagist.org/packages/tighten/existing_package.json' => Http::response([
+                'package' => [
+                    'downloads' => [
+                        'total' => 1000,
+                        'monthly' => 100,
+                    ],
+                ],
+            ]),
+        ]);
+
+        $this->mockGithubClient('tighten', 'existing_package');
+
+        $project = Project::factory()->create([
+            'namespace' => 'tighten',
+            'name' => 'existing_package',
+            'packagist_name' => null,
+        ]);
+        $maintainer = Maintainer::factory()->create([
+            'github_username' => 'Baz',
+        ]);
+        $project->maintainers()->attach($maintainer);
+
+        $this->artisan('stats:fetch')->assertSuccessful();
+        $project->refresh();
+
+        $this->assertEquals($project->downloads_total, 1000);
+        $this->assertEquals($project->downloads_last_30_days, 100);
+
+        $this->assertEquals($project->issues_count, 0);
+        $this->assertEquals($project->pull_requests_count, 0);
+        $this->assertEquals($project->issues, collect([]));
+        $this->assertEquals($project->pull_requests, collect([]));
+
+        $this->assertEquals($project->is_hidden, false);
+        $this->assertNotEmpty(Cache::get('projects'));
+    }
+
+    /** @test */
+    public function fetch_stats_does_not_overwrite_packagist_stats(): void
```

Comment:
If I'm understanding this test correctly, it means a failed request doesn't overwrite our local stats, right? If so, I'd change the method name to indicate this:

```suggestion
    public function failed_stats_fetch_does_not_overwrite_packagist_stats(): void
```